scripts/deploy/github/kfp-readiness/wait_for_pods.py

```diff
@@ -41,8 +51,10 @@ def check_pods(calm_time=10, timeout=600, retries_after_ready=5):
         current_statuses = get_pod_statuses()
 
         logging.info("Checking pod statuses...")
-        for pod_name, (pod_status, ready, total) in current_statuses.items():
+        for pod_name, (pod_status, ready, total, waiting_messages) in current_statuses.items():
             logging.info(f"Pod {pod_name} - Status: {pod_status}, Ready: {ready}/{total}")
+            for waiting_msg  in waiting_messages:
+                logging.info(waiting_msg)
```

Comment:
Perhaps should provide a context message here before dumping the list of waiting messages, even if it's just a static "Reasons for Pending/Not Ready:" header line.  Going off of the example log posted it took a second to realize the `CONTAINER` lines were related to the previous pod status line before it

Reply:
Took the alternative in your comment below, thanks for the suggestion!